tests/functional/admin/test_admin_agent.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (C) 2019 OpenIO SAS
+
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+import unittest
+import os
+import multiprocessing as mp
+from BaseHTTPServer import HTTPServer
+import requests
+from oio.admin.agent import OioAdminAgent, BaseAdminAgentHandler
+from tests.utils import DotDict, FakeBlobMoverSlow, \
+    FakeBlobRebuilderSlow, FakeMeta2MoverSlow, FakeCsClientFactory
+
+STATUS_JOB_TERMINATED = 1
+process = None
+
+
+def make_handler(args):
+    class OioAdminAgentHandler(BaseAdminAgentHandler):
+        agent = OioAdminAgent(args)
+    return OioAdminAgentHandler
+
+
+class TestOioAdminAgentHandler(unittest.TestCase):
+
+    def setUp(self):
+        self.handler = make_handler(DotDict(
+            host="node2",
+            namespace="OPENIO",
+            log_level="INFO",
+            log_facility="local0",
+            log_syslog_prefix="OIO,OPENIO,oio-mover-agent,0",
+            log_address="/dev/log",
+            quiet=False,
+            ctx=dict(
+                conscience=FakeCsClientFactory()(),
+                blob_mover=FakeBlobMoverSlow,
+                blob_rebuilder=FakeBlobRebuilderSlow,
+                meta2_mover=FakeMeta2MoverSlow,
+            )
+        ))
+        httpd = HTTPServer(('localhost', 0), self.handler)
+        self.url = "http://localhost:%s/" % httpd.server_port
+
+        def start(httpd):
+            httpd.serve_forever()
+        global process
```

Comment:
Why global since it doesn't use anywhere else only in tearDown ?
Maybe you should register it in `setUpClass` ?